core/autoscaling.go

```diff
@@ -893,34 +893,27 @@ func (a *autoScalingGroup) attachSpotInstance(spotInstanceID *string) error {
 	return nil
 }
 
-// Terminates an on-demand instance from the group,
-// but only after it was detached from the autoscaling group
-func (a *autoScalingGroup) detachAndTerminateOnDemandInstance(
+// Terminates an on-demand instance from the group using the
+// TerminateInstanceInAutoScalingGroup api call.
+func (a *autoScalingGroup) terminateInstanceInAutoScalingGroup(
 	instanceID *string) error {
 	logger.Println(a.region.name,
 		a.name,
-		"Detaching and terminating instance:",
+		"Terminating instance:",
 		*instanceID)
-	// detach the on-demand instance
-	detachParams := autoscaling.DetachInstancesInput{
-		AutoScalingGroupName: aws.String(a.name),
-		InstanceIds: []*string{
-			instanceID,
-		},
+	// terminate the on-demand instance
+	terminateParams := autoscaling.TerminateInstanceInAutoScalingGroupInput{
+		InstanceId:                     instanceID,
 		ShouldDecrementDesiredCapacity: aws.Bool(true),
```

Comment:
Did you test this against groups currently running at minimum capacity?

Will the termination still happen if the capacity can't be decreased because it's already at minimum?

Reply:
I just tested it and it worked. Right before the terminate action, it's attaching the spot instance so it should be 1 above the minimum count.
```
2018/04/30 11:42:18 autoscaling.go:313: us-west-2 Attaching spot instance i-0b42ef9ec5d5f6c0d to ecs
2018/04/30 11:42:18 autoscaling.go:424: ecs Retrieving instance details for  i-0b42ef9ec5d5f6c0d
2018/04/30 11:42:18 autoscaling.go:431: ecs i-0b42ef9ec5d5f6c0d is in the availability zone us-west-2b looking for an on-demand instance there
2018/04/30 11:42:18 autoscaling.go:444: ecs found on-demand instance i-010f042d7a7968a0c replacing with new spot instance i-0b42ef9ec5d5f6c0d
2018/04/30 11:42:18 autoscaling.go:900: us-west-2 ecs Terminating instance: i-010f042d7a7968a0c
```

Minimum and desired are at 5. After running the lambda, the number of instances goes to 6 from 5, while minimum and desired are still at 5. The terminating instance now executes its lifecycle hook as normal.